nvaccess/nvda
# Browse mode with auto focus focusable elements disabled: Focus focusable ancestor at caret when forcing focus mode with NVDA+space.

### Link to issue number:
Addresses issue mentioned in https://github.com/nvaccess/nvda/pull/9511#issuecomment-497561763 and https://github.com/nvaccess/nvda/pull/11190#issuecomment-633310976.

### Summary of the issue:
When enabling (forcing) focus mode with NVDA+space, NVDA doesn't set focus to the element under the cursor with auto focus focusable elements disabled.

### Description of how this pull request fixes the issue:
Switching to focus mode with NVDA+space sets disableAutoPassThrough to True on the TreeInterceptor. Previously, this was just a variable. Now, it's a property which calls `_focusLastFocusableObject()` if appropriate.

### Testing performed:
STR from https://github.com/nvaccess/nvda/pull/9511#issuecomment-497561763. With automatically set system focus to focusable elements disabled:

1. Go to google.com.
2. Tab to the Google Search button.
3. Press shift+e to quick nav back to the search box.
4. Press NVDA+space to force focus mode.
5. Press NVDA+tab. 
    - Expected: "Search combo box ..."
    - Actual: "Google Search button"

### Known issues with pull request:
None.

### Change log entry:
Bug fixes:
`- In browse mode with Automatically set system focus to focusable elements disabled, switching to focus mode with NVDA+space now focuses the element under the caret.`
